tests/test_generic.py

```diff
@@ -1256,3 +1256,11 @@ def test_contentstream_arrayobject_containing_nullobject(caplog):
     content_stream = ContentStream(stream=input_stream, pdf=None)
     assert content_stream.get_data() == b"Hello World!\n"
     assert caplog.text == ""
+
+
+def test_issue_3419_links_without_slash_d_can_be_copied():
+    reader = PdfReader(RESOURCE_ROOT / "issue-3419.pdf")
```

Comment:
Please avoid adding more files to the resources directory to not bloat this repository. The preferred way is to either download the file on the fly or to add it to [example_files.yaml](https://github.com/py-pdf/pypdf/blob/main/tests/example_files.yaml).

Another alternative would be to add it to https://github.com/py-pdf/sample-files, although I tend to prefer more general documents there.